src/crates-and-source-files.md

```diff
@@ -100,6 +83,48 @@ fn main() {
 }
 ```
 
+## Preludes and `no_std`
+
+All crates have a *prelude* that automatically inserts a [use declaration] into
+each [module] and an [`extern crate]` into the crate root module. By default,
```

Comment:
This is not how it works and there are observable differences.
(Will explain later, need to go now.)

Reply:
I'm still blocked on this explanation.

Reply:
Sorry, I completely forgot about this.

Reply:
Prelude names are not placed in any modules with `use` declarations, they are just added into scope.
So you can write `Vec<T>` in any module, but not `self::Vec<T>` - ther's no `Vec` in any module.

Reply:
If we look at some other names that are placed into scope rather than into modules
```rust
{
    let x = 10;
    {
        let x = 11;
        {
            let x = 12;
            {
                x
            }
        }
    }
}
```
then we can extend this hierarchy with one more outer level and place prelude names there, that's the effect of std prelude injection.

A number of other names behaves in the same way, e.g. built-in types, or macros imported with `#[macro_use] extern crate ...`, or extern crates with `feature(extern_prelude)` (that's currently being stabilized in https://github.com/rust-lang/rust/pull/54404),